src/nvim/api/buffer.c

```diff
@@ -639,7 +638,9 @@ Boolean nvim_buf_is_valid(Buffer buffer)
     FUNC_API_SINCE(1)
 {
   Error stub = ERROR_INIT;
-  return find_buffer_by_handle(buffer, &stub) != NULL;
+  Boolean ret = find_buffer_by_handle(buffer, &stub) != NULL;
+  xfree(stub.msg);
```

Comment:
I would actually suggest using `api_free_error(Error)` or `api_clear_error(Error *)`, would be useful if we find that error handling needs to be optimized: note that e.g. Python is completely fine with EAFP approach which suggests that errors are common and expected.